app/components/progress-updater.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+class ProgressUpdater {
+
+  constructor() {
+
+    this.state = {
+      playing: false,
+      progressObject: null,
+      progressPosition: '0%',
+      subjectViewer: null
+    };
+
+    this.progressUpdateListener = this.handleProgressUpdate.bind(this);
+    this.progressMarkerRenderer = this.renderProgressMarker.bind(this);
+    this.progressMonitoredObject = this.registerProgressingObject.bind(this);
+  }
+
+  setSubjectViewer(subjectViewer){
+    this.state.subjectViewer = subjectViewer;
+  }
+
+  updateSubjectViewer(){
+    if(this.state.subjectViewer){
+      this.state.subjectViewer.forceUpdate();
```

Comment:
Just to expand on that, I was thinking something like `this.props.onProgressChange()` would be better, where the callback is passed down as a prop, and triggers a state change in the parent, which can then be passed down to other children.